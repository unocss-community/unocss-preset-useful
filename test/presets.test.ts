import { createGenerator } from 'unocss'
import { presetUseful } from 'unocss-preset-useful'
import { describe, expect, it } from 'vitest'

describe('preseUseful presets', () => {
  it('mutiple preset', async () => {
    const uno = await createGenerator({
      presets: [
        presetUseful({
          typography: true,
        }),
      ],
    })
    const rawPresets = uno.config.presets
    const sortedPresets = [
      ...rawPresets.filter(p => p.enforce === 'pre'),
      ...rawPresets.filter(p => !p.enforce),
      ...rawPresets.filter(p => p.enforce === 'post'),
    ]
    expect(sortedPresets.map(i => i.name)).toMatchInlineSnapshot(`
      [
        "@unocss/preset-icons",
        "unocss-preset-useful",
        "@unocss/preset-uno",
        "@unocss/preset-attributify",
        "@unocss/preset-typography",
      ]
    `)

    const { css } = await uno.generate('text-center prose', { preflights: false })
    expect(css).toMatchSnapshot()
  })
})
