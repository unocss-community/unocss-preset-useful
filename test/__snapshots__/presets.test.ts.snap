// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`preseUseful presets > mutiple preset 2`] = `
"/* layer: typography */
.prose{color:var(--un-prose-body);max-width:65ch;}
/* layer: shortcuts */
.scrollbar{--scrollbar-track:#f5f5f5;--scrollbar-thumb:#ddd;--scrollbar-width:8px;--scrollbar-height:8px;--scrollbar-track-radius:4px;--scrollbar-thumb-radius:4px;overflow:auto;}
.scrollbar::-webkit-scrollbar{width:var(--scrollbar-width);height:var(--scrollbar-height);}
.scrollbar::-webkit-scrollbar-thumb{background-color:var(--scrollbar-thumb);}
.scrollbar::-webkit-scrollbar-track{background-color:var(--scrollbar-track);}
/* layer: default */
.text-center{text-align:center;}"
`;
